### ISSUE TITLE
failed to build v22.2.8 from source

### ISSUE THREAD

**Issue by davepacheco** [cockroachdb/cockroach#101880]
**Describe the problem**

I'm trying to build v22.2.8 from source with "make" (which I know is deprecated, but I believe it's still supported?).  I'm getting this error:

```
src/transactionDetails/transactionDetails.stories.tsx(34,6): error TS2741: Property 'isDataValid' is missing in type '{ timeScale: TimeScale; transactionFingerprintId: string; transaction: { stats_data: { statement_fingerprint_ids: Long[]; app: string; stats: { count: Long; max_retries: Long; ... 7 more ...; rows_written: { ...; }; }; aggregated_ts: Timestamp; transaction_fingerprint_id: Long; aggregation_interval: { ...; }; }; }; ...' but required in type 'Readonly<TransactionDetailsProps>'.
src/transactionDetails/transactionDetails.stories.tsx(52,6): error TS2741: Property 'isDataValid' is missing in type '{ timeScale: TimeScale; transactionFingerprintId: string; transaction: null; isLoading: true; statements: undefined; nodeRegions: { "1": string; "2": string; "3": string; "4": string; }; isTenant: false; ... 8 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionDetailsProps>'.
src/transactionDetails/transactionDetails.stories.tsx(70,6): error TS2741: Property 'isDataValid' is missing in type '{ timeScale: TimeScale; transactionFingerprintId: undefined; transaction: undefined; isLoading: false; statements: undefined; nodeRegions: { "1": string; "2": string; "3": string; "4": string; }; ... 10 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionDetailsProps>'.
src/transactionDetails/transactionDetails.stories.tsx(90,8): error TS2741: Property 'isDataValid' is missing in type '{ timeScale: TimeScale; transactionFingerprintId: string; transaction: undefined; isLoading: false; statements: ICollectedStatementStatistics[]; ... 10 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionDetailsProps>'.
src/transactionsPage/transactionsPage.stories.tsx(38,6): error TS2741: Property 'isReqInFlight' is missing in type '{ columns: string[]; data: IStatementsResponse; timeScale: TimeScale; filters: Filters; nodeRegions: { [nodeId: string]: string; }; hasAdminRole: true; onFilterChange: (...args: any[]) => void; ... 11 more ...; isDataValid: true; }' but required in type 'Readonly<TransactionsPageProps>'.
src/transactionsPage/transactionsPage.stories.tsx(60,8): error TS2741: Property 'isReqInFlight' is missing in type '{ isDataValid: true; columns: string[]; data: IStatementsResponse & { transactions: any[]; statements: any[]; }; timeScale: TimeScale; filters: Filters; ... 13 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionsPageProps>'.
src/transactionsPage/transactionsPage.stories.tsx(89,8): error TS2741: Property 'isReqInFlight' is missing in type '{ columns: string[]; isDataValid: true; data: IStatementsResponse & { transactions: any[]; statements: any[]; }; timeScale: TimeScale; filters: Filters; ... 13 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionsPageProps>'.
src/transactionsPage/transactionsPage.stories.tsx(113,8): error TS2741: Property 'isReqInFlight' is missing in type '{ columns: string[]; isDataValid: true; data: undefined; timeScale: TimeScale; filters: Filters; nodeRegions: { [nodeId: string]: string; }; hasAdminRole: true; onFilterChange: (...args: any[]) => void; ... 10 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionsPageProps>'.
src/transactionsPage/transactionsPage.stories.tsx(136,8): error TS2741: Property 'isReqInFlight' is missing in type '{ columns: string[]; isDataValid: true; data: undefined; timeScale: TimeScale; error: RequestError; filters: Filters; nodeRegions: { [nodeId: string]: string; }; hasAdminRole: true; ... 11 more ...; match: { ...; }; }' but required in type 'Readonly<TransactionsPageProps>'.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "build:typescript" exited with 2.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [Makefile:370: pkg/ui/cluster-ui/dist/main.js] Error 1
```

**To Reproduce**

Here's roughly what I did: I stared with an Ubuntu 22.04 LTS instance in AWS.  (I used ami-0fcf52bcf5db7b003, instance type c5.4xlarge, with a 1024GiB gpt2 root volume.)

```
$ sudo apt-get update
$ sudo apt-get install libncurses-dev build-essential
$ sudo apt-get install nodejs
$ sudo apt-get install npm 
$ sudo apt-get install yacc 
$ sudo npm install -g yarn
$ sudo apt-get install autoconf
$ sudo apt-get install cmake
$ mkdir install
$ cd install
$ curl -LO https://go.dev/dl/go1.19.8.linux-amd64.tar.gz
$ tar xzf ../go1.19.8.linux-amd64.tar.gz
$ export GOPATH=$HOME/go
$ mkdir -p $(go env GOPATH)/src/github.com/cockroachdb
$ cd $(go env GOPATH)/src/github.com/cockroachdb
$ git clone https://github.com/cockroachdb/cockroach.git
$ cd cockroach
$ git checkout v22.2.8
$ make buildoss BUILDTYPE=release
```

I originally saw this on illumos, not Linux, and with some local patches.  Here I've reproduced it with stock 22.2.8 on Linux.  I see this with both the normal build (`make build BUILDTYPE=release`) and OSS build (see above) on both illumos and Linux.

Full build output: https://gist.github.com/davepacheco/86fe609d016a804a303240f155c9243c

Jira issue: CRDB-27171

**Comment by rickystewart** [comment]
@sjbarag Do you have any insight here?

**Comment by sjbarag** [comment]
Looks like the root-cause is some skew between our Bazel build and our Make build. The make build attempts to type-check some `.stories.tsx` files that are used for [a UI component demo environment](https://storybook.js.org), but Bazel ignores those. I can get a PR up to fix that pretty easily — I'm pretty sure this also happens in `master`.

**Comment by davepacheco** [comment]
That's good news.  Once there's a PR, we can patch it locally to work around it and move to 22.2.8.

**Comment by davepacheco** [comment]
Is there an update on this one?  Thanks!

**Comment by davepacheco** [comment]
Yes, unfortunately last I checked Bazel is not supported on our platform (illumos) yet.  I know we need to get there eventually, but I gather it's going to be a fair bit of work.

We're approaching the point where it'll be too risky for us to upgrade to v22.2 for our first release.  That is, we can't update to v22.2.8 because of this issue and the technical advisories affecting earlier v22.2 releases look too risky.  If it's a simple fix for this one, and it sounds like "make" is supposed to be supported on this release train, a fix for this sure would be appreciated!

**Comment by rickystewart** [comment]
@davepacheco The next v22.2 release should have this problem solved (assuming it has the commit `c13e298a60605de4550b2d9bc3058476daa43c9c`). Let me know if there are any further issues.

**Comment by davepacheco** [comment]
Thanks!
